__tests__/__packages__/cli-test-utils/src/environment/TempTestProfiles.ts

```diff
@@ -0,0 +1,224 @@
+     * @param {string} profileName - the name of the profile to delete
+     * @returns {Promise<string>} promise that resolves to the name of the created profile on success
+     * @throws errors if the profile delete fails
+     */
+    private static async deleteV1Profile(testEnvironment: ITestEnvironment<any>, profileType: string, profileName: string): Promise<string> {
+        const deleteProfileScript = this.SHEBANG + `${this.BINARY_NAME} profiles delete ${profileType} ${profileName} --force`;
+        const scriptPath = testEnvironment.workingDir + "_delete_profile_" + profileName;
+        await IO.writeFileAsync(scriptPath, deleteProfileScript);
+        const output = runCliScript(scriptPath, testEnvironment, []);
+        if (output.status !== 0 || !isStderrEmptyForProfilesCommand(output.stderr)) {
+            throw new ImperativeError({
+                msg: "Deletion of " + profileType + " profile '" + profileName + "' failed! You should delete the script: '" + scriptPath + "' " +
+                    "after reviewing it to check for possible errors. Stderr of the profile create command:\n" + output.stderr.toString()
+                    + TempTestProfiles.GLOBAL_INSTALL_NOTE
+            });
+        }
+        this.log(testEnvironment, `Deleted ${profileType} V1 profile '${profileName}'. Stdout from deletion:\n${output.stdout.toString()}`);
+        IO.deleteFile(scriptPath);
+        return profileName;
+    }
+
+    /**
+     * Helper to delete a temporary team config profile
+     * @param {ITestEnvironment} testEnvironment - the test environment with env and working directory to use for output
+     * @param {string} profileType - the type of profile e.g. zosmf to
+     * @param {string} profileName - the name of the profile to delete
+     * @returns {Promise<string>} promise that resolves to the name of the created profile on success
+     * @throws errors if the profile delete fails
+     */
+    private static async deleteV2Profile(testEnvironment: ITestEnvironment<any>, profileType: string, profileName: string): Promise<string> {
+        // Load global config layer
+        const config = await Config.load(this.BINARY_NAME, { homeDir: testEnvironment.workingDir });
+        config.api.layers.activate(false, true);
+
+        // Remove profile from config JSON
+        config.delete(config.api.profiles.expandPath(profileName));
+        if (config.layerActive().properties.defaults[profileType] === profileName) {
+            config.delete(`defaults.${profileType}`);
+        }
+        await config.api.layers.write();
+
+        this.log(testEnvironment, `Deleted ${profileType} V2 profile '${profileName}'`);
+        return profileName;
+    }
+
+    /**
+     * log a message to a file in the working directory
+     */
+    private static log(testEnvironment: ITestEnvironment<any>, message: string) {
+        fs.appendFileSync(testEnvironment.workingDir + "/" + TempTestProfiles.LOG_FILE_NAME, message + "\n");
```

Comment:
Probably not worth the effort but have we thought of using streams instead of regular blocking I/O operations?